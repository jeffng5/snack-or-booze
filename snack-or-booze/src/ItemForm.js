import React, { useState } from 'react'



function ItemForm() {
    const [item, setItem] = useState("")

    const handleSubmit = (e) => {
        e.preventDefault();
        setItem(e.target.value)
    
    }

    const handleChange = (e) => {
        setItem(e.target.value)
        console.log(e.target.value)
    }

return (

    <form>
        <label>Food Item:</label>
        <input type="text" id="food name" name="food name" onChange={handleChange}></input>
        <label>Description:</label>
        <input type="text" id="description" name= "description" onChange={handleChange}></input>
        <label>Recipe:</label>
        <input type="text" id="recipe" name= "recipe" onChange={handleChange}></input>
        <label>Serve:</label>
        <input type="text" id="serve" name= "serve" onChange={handleChange}></input>
        <button onClick={handleSubmit}>Add Item</button>
    </form>

)

}


export default ItemForm