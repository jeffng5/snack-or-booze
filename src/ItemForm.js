import React, { useState } from 'react'
import './App.css'


const ItemForm = () =>  {
    const initialState =[]
    const [formData, setFormData] = useState(initialState)

    
    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({...formData, [name]: value}))
     
     }
    
    const handleSubmit = (e) => {
        e.preventDefault();
        setFormData(formData.food, formData.description, formData.recipe, formData.serve)
        setFormData(initialState)
    
    }


return (
<>
    <form>
        <label>Food Item:</label>

        <div>
        <input type="text" id="food" name="food" value={formData.food} placeholder= 'item' onChange={handleChange}></input>
        </div>
        <div>
        <input type="text" id="description" name= "description" value={formData.description} placeholder= 'description' onChange={handleChange}></input>
        </div>
        <div>
        <input type="text" id="recipe" name= "recipe" value={formData.recipe} placeholder = 'recipe' onChange={handleChange}></input>
        </div>
        <div>
        <input type="text" id="serve" name= "serve" value={formData.serve} placeholder ='serve' onChange={handleChange}></input>
        </div>
        <button onClick={handleSubmit}>Add Item</button>
    </form>
    </>
)

}


export default ItemForm