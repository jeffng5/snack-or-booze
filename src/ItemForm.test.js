import React from 'react';
import { render } from '@testing-library/react';
import ItemForm from './ItemForm';

//smoke test
it('renders without crashing', function() {
  render(<ItemForm />);
})

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<ItemForm />);
  expect(asFragment()).toMatchSnapshot();
});