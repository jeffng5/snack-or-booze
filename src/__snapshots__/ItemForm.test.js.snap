// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label>
      Food Item:
    </label>
    <div>
      <input
        id="food"
        name="food"
        placeholder="item"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        id="description"
        name="description"
        placeholder="description"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        id="recipe"
        name="recipe"
        placeholder="recipe"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        id="serve"
        name="serve"
        placeholder="serve"
        type="text"
        value=""
      />
    </div>
    <button>
      Add Item
    </button>
  </form>
</DocumentFragment>
`;
